include/cpp-statsd-client/StatsdClient.hpp

```diff
@@ -40,7 +62,8 @@ class StatsdClient {
     StatsdClient(const std::string& host,
                  const uint16_t port,
                  const std::string& prefix,
-                 const uint64_t batchsize = 0) noexcept;
+                 const uint64_t batchsize = 0,
+                 const uint64_t sendInterval = 1000) noexcept;
```

Comment:
so now we allow the caller to configure the interval at which queued stats batches are sent. this implies that you are batching. here we default to the previously hardcoded value of 1 second. if you set the value to 0 then automatic sending via the background thread is disabled and the caller must manually flush the batches themselves.